src/main/java/bob/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,119 @@
+package bob.commands;
+
+import java.io.IOException;
+
+import bob.data.task.Deadline;
+import bob.data.task.Tasklist;
+import bob.exceptions.BobEmptyTaskException;
+import bob.exceptions.BobInvalidDateAndTimeException;
+import bob.storage.Storage;
+import bob.ui.Ui;
+
+/**
+ * Adds a deadline to the Bob's tasklist.
+ */
+public class DeadlineCommand extends Command {
+    private final String input;
+
+    /**
+     * Creates a deadline command.
+     *
+     * @param input User input.
+     * @throws BobEmptyTaskException If there is no description for deadline.
+     */
+    public DeadlineCommand(String input) throws BobEmptyTaskException {
+        //Removes all whitespaces and checks if input is empty
+        if (input.replaceAll("\\s+", "").length() == 0) {
```

Comment:
You can consider using trim() which should achieve similar outcome.

Reply:
I see, thanks for the suggestion I will probably use trim from now on!